IdentityCore/src/requests/sdk/msal/MSIDDefaultTokenResponseValidator.m

```diff
@@ -43,13 +44,18 @@ - (BOOL)validateTokenResult:(MSIDTokenResult *)tokenResult
      we'd like to throw an error and specify which scopes were granted and which ones not
      */
 
-    NSOrderedSet *grantedScopes = [tokenResult.tokenResponse.scope msidScopeSet];
+    NSMutableOrderedSet *grantedScopes = [[tokenResult.tokenResponse.scope msidScopeSet] mutableCopy];
+    NSOrderedSet *oidcScopes = [oidcScope msidScopeSet];
 
     if (![configuration.scopes isSubsetOfOrderedSet:grantedScopes])
     {
         if (error)
         {
             NSMutableDictionary *additionalUserInfo = [NSMutableDictionary new];
+            
+            MSID_LOG_ERROR_CORR(correlationID, @"Server returned less scopes than requested, granted scopes: %@", grantedScopes);
+            // Remove oidc scopes.
```

Comment:
nit: can we log what ones we are removing?

Reply:
done